src/sqlfluff/dialects/dialect_athena.py

```diff
@@ -0,0 +1,541 @@
+                        Ref("EqualsSegment"),
+                        Ref("LiteralGrammar"),
+                        optional=True,
+                    ),
+                )
+            )
+        ),
+    ),
+    BackQuotedIdentifierSegment=NamedParser(
+        "back_quote",
+        CodeSegment,
+        name="quoted_identifier",
+        type="identifier",
+    ),
+)
+
+athena_dialect.replace(
+    QuotedLiteralSegment=OneOf(
+        NamedParser("single_quote", CodeSegment, name="quoted_literal", type="literal"),
+        NamedParser("double_quote", CodeSegment, name="quoted_literal", type="literal"),
+        NamedParser("back_quote", CodeSegment, name="quoted_literal", type="literal"),
+    ),
+    LiteralGrammar=OneOf(
+        Ref("QuotedLiteralSegment"),
+        Ref("NumericLiteralSegment"),
+        Ref("BooleanLiteralGrammar"),
+        Ref("QualifiedNumericLiteralSegment"),
+        # NB: Null is included in the literals, because it is a keyword which
+        # can otherwise be easily mistaken for an identifier.
+        Ref("NullLiteralSegment"),
+        Ref("DateTimeLiteralGrammar"),
+        Sequence(Ref("SimpleArrayTypeGrammar"), Ref("ArrayLiteralSegment")),
+    ),
+    SimpleArrayTypeGrammar=Ref.keyword("ARRAY"),
+    TrimParametersGrammar=Nothing(),
+    SingleIdentifierGrammar=ansi_dialect.get_grammar("SingleIdentifierGrammar").copy(
+        insert=[
+            Ref("BackQuotedIdentifierSegment"),
+        ]
+    ),
+    BinaryOperatorGrammar=OneOf(
+        Ref("ArithmeticBinaryOperatorGrammar"),
+        Ref("StringBinaryOperatorGrammar"),
+        Ref("BooleanBinaryOperatorGrammar"),
+        Ref("ComparisonOperatorGrammar"),
+        # Add arrow operators for functions (e.g. filter)
+        Ref("RightArrowOperator"),
+    ),
+)
+
+
+class PrimitiveTypeSegment(BaseSegment):
+    """Primitive data types.
+
+    Since DDL is based on Hive and DML based on Prestodb this class has
+    primitives that may not work on specific situations
+    - Hive: https://cwiki.apache.org/confluence/display/hive/languagemanual+types
+    - PrestoDb: https://prestodb.io/docs/0.217/language/types.html
+    """
+
+    type = "primitive_type"
+    match_grammar = OneOf(
+        "BOOLEAN",
+        "TINYINT",
+        "SMALLINT",
+        "INTEGER",
+        "INT",  # Hive only
+        "BIGINT",
+        "REAL",  # PrestoDB only
+        "FLOAT",
+        Sequence("DOUBLE", Ref.keyword("PRECISION", optional=True)),
+        Sequence(
+            "DECIMAL",
+            Bracketed(
+                Ref("NumericLiteralSegment"),
+                Ref("CommaSegment"),
+                Ref("NumericLiteralSegment"),
+                optional=True,
+            ),
+        ),
+        "NUMERIC",  # Hive only
+        "STRING",  # Hive only
+        "VARCHAR",
+        "CHAR",
+        "VARBINARY",  # PrestoDB only
+        "JSON",  # PrestoDB only
+        "DATE",
+        "TIMESTAMP",
+        "TIMESTAMP WITH TIME ZONE",  # PrestoDB only
```

Comment:
Does this work? Usually things like this are modeled as a sequence of keywords, not a single string.

Reply:
Thanks for point it out. 

Just added a sequence for these types and created a specific fixture testing it.